slack/workflows/statuspage/__init__.py

```diff
@@ -0,0 +1,44 @@
+import statuspageio
+
+import logging
+logger = logging.getLogger(name="statuspage init")
+
+from .action_handler import *
+from .incident_command import *
+from .dialog_handler import *
+from .constants import *
+
+from .connections import set_status_page_conn
+
+from slack.models import Workflow
+from slack.decorators import incident_command, remove_incident_command
+from slack.decorators import action_handler, remove_action_handler, ActionContext
+from slack.decorators import dialog_handler, remove_dialog_handler
+from slack.decorators.workflow import register_workflow, WorkFlowClass
+
+
+@register_workflow('statuspage')
+class Statuspage(WorkFlowClass):
+    configuration_values = ['STATUSPAGEIO_API_KEY', 'STATUSPAGEIO_PAGE_ID']
+
+    def enabled(self, workflow: Workflow):
+        params = dict((x.name, x.value) for x in workflow.parameters.all())
+
+        required = ['STATUSPAGEIO_API_KEY', 'STATUSPAGEIO_PAGE_ID']
+        if not all([ params[x] for x in params.keys() & required]):
+            logger.warn("Not all required parameters are set for statuspage workflow.")
+            return
+
+        set_status_page_conn(statuspageio.Client(api_key=params['STATUSPAGEIO_API_KEY'],
+                                          page_id=params['STATUSPAGEIO_PAGE_ID']))
+
+        incident_command(STATUS_PAGE_COMMANDS, helptext='Update the statuspage for this incident')(handle_statuspage)
+        action_handler(OPEN_STATUS_PAGE_DIALOG)(handle_open_status_page_dialog)
```

Comment:
Using decorators in this way feels a bit awkward.  Can we add an `add_incident_command` function, for example?

Reply:
Made a change to all decorators so they can be used either way. Take a look.